compiler/semexprs.nim

```diff
@@ -1780,6 +1800,13 @@ proc semAsgn(c: PContext, n: PNode; mode=asgnNormal): PNode =
       # unfortunately we need to rewrite ``(x, y) = foo()`` already here so
       # that overloading of the assignment operator still works. Usually we
       # prefer to do these rewritings in transf.nim:
+      let value = n.lastSon
+      if value.isAtom and value.kind != nkTupleConstr:
+        let v = semExpr(c, value, {efNoError, efFromUnpack})
+        if v.typ.kind != tyTuple:
+          if not isSubscrible(c, value):
+            localError(c.config, value.info, "cannot unpack '$1'" % renderTree(value, {renderNoComments}))
+            return errorNode(c, value)
       return semStmt(c, lowerTupleUnpackingForAsgn(c.graph, n, c.idgen, c.p.owner), {})
```

Comment:
What about a patch for `lowerTupleUnpackingForAsgn`?

Reply:
I don't think so, `lowerTupleUnpackingForAsgn` does exactly as it's name, here few lines looks?fine to me